Author: Svante Signell <svante.signell@telia.com>
Description: Enable build of all targets on GNU/Hurd (Closes: #643649).

diff -Naurp dmidecode.orig/Makefile dmidecode/Makefile
--- dmidecode.orig/Makefile	2011-09-28 06:23:12.627245890 +0200
+++ dmidecode/Makefile	2011-09-28 13:02:12.089955791 +0200
@@ -39,7 +39,7 @@ INSTALL_PROGRAM := $(INSTALL) -m 755
 RM              := rm -f
 
 # BSD make provides $MACHINE, but GNU make doesn't
-MACHINE ?= $(shell uname -m 2>/dev/null)
+MACHINE ?= $(shell uname -m 2>/dev/null | cut - -d- -f 1)
 
 # These programs are only useful on x86
 PROGRAMS-i386 := biosdecode ownership vpddecode
